Quote all fields in exported CSV files

ronaldfw opened the pull request:
Fields in our exported CSV files may contain commas, for example,
strings that represent functions with arguments. To ensure that these
CSV files can be correctly imported in spreadsheet applications, values
for these fields need to be quoted. We also escape any existing quotes
in values. Added a unit test to ensure correctness of the formatting 
code. 

Tested: Exported CSV file from sampled callstack, imported in Google
Sheets. Unit test.
Bug: http://b/181847321

beckerhe commented on src/OrbitGl/DataView.cpp:
```diff
@@ -102,10 +102,19 @@ ErrorMessageOr<void> DataView::ExportCSV(const std::filesystem::path& file_path)
+  // https://tools.ietf.org/html/rfc4180.
+  auto append_with_quotes = [](std::string* s, const std::string_view value) {
+    s->append("\"");
+    s->append(value);
```
AFAIK, quotes in `value` need to be escaped for this to work. Or are we sure that quotes cannot appear in data view fields? If so, this might warrant a comment explaining that.

ronaldfw replied:
Yes, you are right. I was assuming this doesn't occur, but I'll just fix this right now.

ronaldfw replied:
Ok, should be fixed now.

beckerhe reviewed (approved):
Thanks. Otherwise LGTM!

ronaldfw commented:
I've addressed your comment that quotes need to be escaped (with another quote) when quoting the fields. I added a unit test for the CSV formatting function. The CSV formatting code and test could be moved to OrbitBase, but since they are not used anywhere else I figured it would be better to keep it local in DataView. But happy to address any additional comments if you don't agree with the choices here.

beckerhe reviewed (approved):
Very cool. Thanks. LGTM!

dimitry- commented on src/OrbitGl/DataView.cpp:
```diff
@@ -9,6 +9,20 @@
 #include "OrbitBase/File.h"
 #include "OrbitBase/Logging.h"
+#include "absl/strings/str_join.h"
+#include "absl/strings/str_split.h"
```
nit: we use `<>` include style for external to Orbit headers

ronaldfw replied:
Ah, thanks, will fix right away!

ronaldfw replied:
Should be fixed now.

dimitry- commented on src/OrbitGl/DataView.cpp:
```diff
@@ -4,12 +4,27 @@
+  std::string result;
+  result.append("\"");
+  std::vector<std::string> split_value = absl::StrSplit(value, "\"");
+  std::string value_with_escaped_quotes = absl::StrJoin(split_value, "\"\"");
```
Maybe `absl::StrReplaceAll({"\"", "\"\""})`?

ronaldfw replied:
Great, that's better, will change it!

dimitry- reviewed (approved):
thanks!
